sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -4283,15 +4292,11 @@ pi_result piEnqueueMemBufferMap(pi_queue Queue, pi_mem Buffer,
   ze_event_handle_t ZeEvent = nullptr;
 
   if (Event) {
-    auto Res = piEventCreate(Queue->Context, Event);
+    auto Res = createEventAndAssociateQueue(Queue, Event,
+                                            PI_COMMAND_TYPE_MEM_BUFFER_MAP,
+                                            ZeCommandList, &ZeEvent, false);
```

Comment:
Please move the queue lock made in line 4333 before here and remove the last argument.

Reply:
Done. Thanks.